kombu/utils/json.py

```diff
@@ -34,13 +35,13 @@ class _DecodeError(Exception):
 class JSONEncoder(_encoder_cls):
     """Kombu custom json encoder."""
 
-    def default(self, o,
-                dates=(datetime.datetime, datetime.date),
-                times=(datetime.time,),
-                textual=(decimal.Decimal, uuid.UUID, DjangoPromise),
-                isinstance=isinstance,
-                datetime=datetime.datetime,
-                text_t=str):
+    def default(self, o: Any,
+                dates: Iterable = (datetime.datetime, datetime.date),
+                times: Iterable = (datetime.time,),
+                textual: Any = (decimal.Decimal, uuid.UUID, DjangoPromise),
+                isinstance: Callable = isinstance,
```

Comment:
Think about it.
<img width="599" alt="image" src="https://user-images.githubusercontent.com/7353520/164030060-6c6b1428-4950-4481-bf14-6466eb3776c4.png">